### DISCUSSION TITLE
Just started, a few questions

### DISCUSSION THREAD

**Discussion by noiselevelzero** [rdz-oss/BattyBirdNET-Pi#35] [General]
I have been running Birdnet pi for a while now.  I just got an Audiomoth and have configured a Pi4 for Batty Birdnet.  I am having some issues, so want to make sure I am missing nothing:

1.  In Bat classifier settings, what do I set so that I detect birds during the day and bats at night.  I have checked "switch to bird detection" and "use lat/long" and saved it multiple times, but after a reboot, the checkbox reappears in "Dusk to Dawn Timer" so it will never detect birds after a reboot.
2. What is the difference between the two audiomoth profiles?
3. Is "Bat Classifier= USA East" correct for me in Virginia to receive both birds and bats during the two times of the day, or should I pick something else?
4. I am unable to play the audio on almost all of my detections. One others, I get a very short (less than a 2nd) burst of audio.  What might cause this?

Thank you for making this available!  We have a Bat house up on our land, and are looking forward to hearing them when the birds shut up for the day!

**Comment by rdz-oss** [comment]
Hi @noiselevelzero ,

if you came this far you will get it to work!
1. You want birds detected during the day and bats at night these are the settings:
<img width="579" alt="Screenshot 2024-07-12 at 11 46 33" src="https://github.com/user-attachments/assets/f21afa26-4c5a-42b5-b0cd-666826002f00">

You set all three!

2. The audiomoth settings are the same. It is a UI thing that once it shows with the complete path and once without.

3. In Virgina you want USA EAST as a bat classifier

4. The audio is played in audible range unless you use the frequency shift button available on the 'Recordings' page. Some browser download the entire audio file before starting to play it (Firefox), others such as Brave or Chrome start playing while downloading.

<img width="198" alt="Screenshot 2024-07-12 at 11 49 56" src="https://github.com/user-attachments/assets/7679fbe5-edc1-4690-b7cc-bc2ce233d1a8">

Hope that helps! Let me know if you get stuck!

Best
Richard

**Reply by noiselevelzero** [reply]
Hi, thanks. I am still having some difficulty. I've been detecting plenty of bats, but I think some of these are false positives because I have been logging upwards of 10K Hoary Bat calls a day.  I am sure I am detecting bats, because I have another audiomoth that is in standalone mode, and I can see the calls.

I also have tried the shift feature, but I've yet to hear a bat call that way.  I can take files from the other audiomoth externally and process them in the audiomoth play app or Audacity, and see many calls that way.  An example is attached.  What I would like is to be able to listen to some of the calls I receive on the Batty Birdnet pi too.

My spectrograms are also useless; I see a very compressed spectrogram which really doesn't let me see calls in any useful way.  Am I doing something wrong with that?

https://github.com/user-attachments/assets/459a4947-9786-4f27-8e5c-f08695b81e3b



Finally, I am seeing backlogs that take hours to clear.  What is the recommended recording and extraction lengths that make this work best?

Thank you!

**Reply by rdz-oss** [reply]
Hi @noiselevelzero 

lets see. I would need some more info to better understand what is broken. A working system with US EAST classifier is here http://73.214.236.157:8088/ so you can get a feeling where your system is off/different. What does the Logging view show?

Getting rid of backlogs
- Set the sampling rate to lowest, ie. 256kHz.
- Set the recording length up, i.e. 14 seconds 
- Change to a Pi 5
- Have a high performance sd card (write operation speed can be a bottleneck)

Spectrograms
- you can compare to those available on http://73.214.236.157:8088  The resolution of x and y axis depend on parameters that can be set in the code. If you make smaller extracts (e.g. 1.125 seconds or so) the resolution is usually sufficient for confirmation of id.

10k hoary bat calls
- indicate that either you have some environmental noise that triggers this or that the input sound stream is not working, i.e. that hoary bats are detected on an empty sound stream. Could you post a snapshot of a spectrogram? Also, What do you see in the currently analyzing spectrogram on the overview page? 

Shift feature
- what is does is slow down the recording and make it 10 times lower in frequency. The behavior also depends on the browser, some need to load the entire file before they start playing it. Download the .wav file and then try to listen to it.

Hope that helps. Kindly post some info on a spectrogram (currently analyzing as well as a 'detection' ) and the LogView output.


BR
Richard

**Reply by rdz-oss** [reply]
On spectrograms - you can use the tool sox ( -X option to say 1000) or audacity to change the x axis resolution to better see the shape of individual calls. The ML algorithm uses entire segments of calls (0.526 seconds) to determine the species - not individual calls.

**Reply by rdz-oss** [reply]
If the issue with Hoary bats persist you can send me about 20-100 files that were triggered by noise. I can add them to the classifier data set and they will be ignored in the future

**Comment by noiselevelzero** [comment]
Hi, Richard, thanks for the suggestions.  I did finally get it working the right way, but that took using one of my Raspberry Pi 5's, and a fully manual installation.  It appears to me that it works, but poorly for me, with a Pi-4 despite having an 8GB model and plenty of storage.

I tried to use the image to build, but found that if I did that I could not log in to the Tools menu for configuration.  I wiped the install off the image at least 4 times and reinstalled using the script, with no luck.  I then uninstalled the software manually rm -r (because the uninstall.sh is not present on the image) and did that perhaps another 3 or 4 times.  Finally, I just discarded using the provided image and loaded the OS from scratch, and then used the script install, which then allowed me to access the BattyBirdnet interface, and it came up and started operating on my Pi5 with Audiomoth.  Now, I can actually see traces like others have posted, which I was not before, and the frequency and time shift works correctly.

You might want to revisit the image you have linked to.  I think there is something wrong with it, otherwise, the manual install, using a Pi5, works very well.

Thank you!
